domain/grant.go

```diff
@@ -22,25 +22,27 @@ const (
 )
 
 type Grant struct {
-	ID               string      `json:"id" yaml:"id"`
-	Status           GrantStatus `json:"status" yaml:"status"`
-	StatusInProvider GrantStatus `json:"status_in_provider" yaml:"status_in_provider"`
-	AccountID        string      `json:"account_id" yaml:"account_id"`
-	AccountType      string      `json:"account_type" yaml:"account_type"`
-	ResourceID       string      `json:"resource_id" yaml:"resource_id"`
-	Role             string      `json:"role" yaml:"role"`
-	Permissions      []string    `json:"permissions" yaml:"permissions"`
-	IsPermanent      bool        `json:"is_permanent" yaml:"is_permanent"`
-	ExpirationDate   *time.Time  `json:"expiration_date" yaml:"expiration_date"`
-	AppealID         string      `json:"appeal_id" yaml:"appeal_id"`
-	Source           GrantSource `json:"source" yaml:"source"`
-	RevokedBy        string      `json:"revoked_by,omitempty" yaml:"revoked_by,omitempty"`
-	RevokedAt        *time.Time  `json:"revoked_at,omitempty" yaml:"revoked_at,omitempty"`
-	RevokeReason     string      `json:"revoke_reason,omitempty" yaml:"revoke_reason,omitempty"`
-	CreatedBy        string      `json:"created_by" yaml:"created_by"` // Deprecated: use Owner instead
-	Owner            string      `json:"owner" yaml:"owner"`
-	CreatedAt        time.Time   `json:"created_at" yaml:"created_at"`
-	UpdatedAt        time.Time   `json:"updated_at" yaml:"updated_at"`
+	ID                      string      `json:"id" yaml:"id"`
+	Status                  GrantStatus `json:"status" yaml:"status"`
+	StatusInProvider        GrantStatus `json:"status_in_provider" yaml:"status_in_provider"`
+	AccountID               string      `json:"account_id" yaml:"account_id"`
+	AccountType             string      `json:"account_type" yaml:"account_type"`
+	ResourceID              string      `json:"resource_id" yaml:"resource_id"`
+	Role                    string      `json:"role" yaml:"role"`
+	Permissions             []string    `json:"permissions" yaml:"permissions"`
+	IsPermanent             bool        `json:"is_permanent" yaml:"is_permanent"`
+	ExpirationDate          *time.Time  `json:"expiration_date" yaml:"expiration_date"`
+	RequestedExpirationDate *time.Time  `json:"requested_expiration_date,omitempty" yaml:"requested_expiration_date,omitempty"`
+	ExpirationDateReason    string      `json:"expiration_date_reason,omitempty" yaml:"expiration_date_reason,omitempty"`
```

Comment:
Why not just

```suggestion
	ExpirationReason    string      `json:"expiration_reason,omitempty" yaml:"expiration_reason,omitempty"`
```

?

Reply:
initially did this, but I think it could be misunderstood as the reason why the grant got expired, wdyt?

Reply:
I think it is okay to use `ExpirationDateReason` for this then.